services/provider/server/server.go

```diff
@@ -135,6 +135,33 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 		return nil, status.Errorf(codes.Internal, "failed to get public key to validate onboarding ticket for consumer %q. %v", req.ConsumerName, err)
 	}
 
+	// get the consumer corresponding to this ticket
+	storageConsumerList := &ocsv1alpha1.StorageConsumerList{}
+	if err := s.client.List(ctx, storageConsumerList, client.InNamespace(s.namespace)); err != nil {
+		klog.Errorf("failed to get storageconsumers in the namespace: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get storageconsumers. %v", err)
+	} else if len(storageConsumerList.Items) < 1 {
+		return nil, status.Errorf(codes.FailedPrecondition, "no storageconsumers exist in the namespace")
+	}
+
+	consumerName := ""
+	for idx := range storageConsumerList.Items {
```

Comment:
Instead of fetching each consumer, getting the secret attached and comparing the onboarding ticket in the secret, why not add the consumer name to the onboarding ticket struct?

Reply:
we could do that after a small discussion as it implies the token will be applicable only to the specific storageconsumer.

Reply:
changed to use an id tying storageconsumer with token.